### DISCUSSION TITLE
Segmentation fault

### DISCUSSION THREAD

**Discussion by anais-mv** [IIC2133-PUC/2024-1#44] [Ayuda]
Hola, no he podido avanzar en la tarea porque tengo un error de Segmentation fault que me tiene estancada. Estoy usando valgrind para ver de dónde viene el error pero me tira a la linea int main() y por lo que tengo entendido los argumentos no tengo que inicializarlos antes, entonces no entiendo de dónde viene mi error u_u

**Comment by benjavicente** [comment]
Hola! Sube lo que llevas a tu repositorio para poder entender lo que está pasando.

**Reply by anais-mv** [reply]
hola! Segmentation fault me dejó de aparecer pero me siguen apareciendo esos errores al correrlo con valgrind, ya subí lo que llevo a mi repositorio.

**Reply by benjavicente** [reply]
Con lo que subiste (ojo con subirlo de manera que compile con `make`), el único error que vi es que no se liberan los struct de archivos, que está incluido en el código base (línea 59 y 60 de `src/ship/main.c`).

Si no es eso, ve si está subido ok todo tu avance y explicita más el error que tienes.
